utils/src/test/java/io/helidon/build/util/MavenVersionsTest.java

```diff
@@ -0,0 +1,154 @@
+                                                    .fallbackVersions(List.of("2.0.0-SNAPSHOT",
+                                                                              "2.0.0-M1",
+                                                                              "2.0.0",
+                                                                              "1.0.0"))
+                                                    .build();
+        assertThat(versions, is(not(nullValue())));
+        assertThat(versions.source(), containsString("fallback"));
+        assertThat(versions.versions(), is(not(nullValue())));
+        assertThat(versions.versions().size(), is(4));
+        assertThat(versions.versions().get(0), is(toMavenVersion("2.0.0")));
+        assertThat(versions.versions().get(1), is(toMavenVersion("2.0.0-SNAPSHOT")));
+        assertThat(versions.versions().get(2), is(toMavenVersion("2.0.0-M1")));
+        assertThat(versions.versions().get(3), is(toMavenVersion("1.0.0")));
+        assertThat(versions.latest(), is(toMavenVersion("2.0.0")));
+    }
+
+    @Test
+    void testHelidonReleases() {
+        final MavenVersions versions = MavenVersions.builder()
+                                                    .artifactGroupId(HELIDON_BOM_GROUP_ID)
+                                                    .artifactId(HELIDON_BOM_ARTIFACT_ID)
+                                                    .build();
+        assertThat(versions, is(not(nullValue())));
+        assertThat(versions.source(), containsString("http"));
+        assertThat(versions.versions(), is(not(nullValue())));
+        assertThat(versions.versions(), is(not(empty())));
+        assertThat(versions.latest(), is(not(nullValue())));
+        assertThat(versions.versions().contains(toMavenVersion("2.0.0-M1")), is(true));
+    }
+
+    @Test
+    void testUnqualifiedHelidonReleases() {
+        final MavenVersions versions = MavenVersions.builder()
+                                                    .filter(MavenVersion.notQualified())
+                                                    .artifactGroupId(HELIDON_BOM_GROUP_ID)
+                                                    .artifactId(HELIDON_BOM_ARTIFACT_ID)
+                                                    .build();
+        assertThat(versions, is(not(nullValue())));
+        assertThat(versions.source(), containsString("http"));
+        assertThat(versions.versions(), is(not(nullValue())));
+        assertThat(versions.versions(), is(not(empty())));
+        assertThat(versions.latest(), is(not(nullValue())));
+        assertThat(versions.versions().contains(toMavenVersion("2.0.0-M1")), is(false));
+    }
+
+    @Test
+    void testLocalHelidonBuilds() {
+        final Path userHome = Paths.get(System.getProperty("user.home"));
+        final Path localRepo = userHome.resolve(".m2/repository");
+        final Path metadataFile = localRepo.resolve("io/helidon/helidon-bom/maven-metadata-local.xml");
```

Comment:
This de-isolates the test, can we mock the local metadata file instead of relying on something to exist in the local repository ?

Reply:
Then there's really no point in the test. I'm not sure what the problem is...?

Reply:
Two problems:.

The test requires something it does not setup, it's not isolated. If we don't want to mock the file then this should be done with an integration test that runs after a `mvn install`.

The test assume that the local repository is a `~/.m2/repository` which is not always true.
Since surefire forks the JVM it's effectively not possible to know the actual local maven repository. Instead this could be done with a dummy project installed in a local repository under the target directory (using the maven invoker plugin as part of the integration-test phase).

Reply:
Maybe you missed the `assumeTrue()` call? It does not require anything; if the file is there, the test runs, if not, it doesn't.

Reply:
I did not miss it.